File: app/assets/javascripts/app/model/CashMutation.js
Ext.define('AM.model.CashMutation', {
  	extend: 'Ext.data.Model',
  	fields: [
    	{ name: 'id', type: 'int' },
			{ name: 'source_class', type: 'string' },
      { name: 'source_id', type: 'int' },
      { name: 'source_code', type: 'string' },
      { name: 'amount', type: 'string' },
      { name: 'mutation_date', type: 'string' },
      { name: 'status', type: 'int' },
      { name: 'status_text', type: 'string' },
      { name: 'cash_bank_name', type: 'string' },
      { name: 'cash_bank_id', type: 'int' },
  	],

	 


   
  	idProperty: 'id' ,proxy: {
			url: 'api/cash_mutations',
			type: 'rest',
			format: 'json',

			reader: {
				root: 'cash_mutations',
				successProperty: 'success',
				totalProperty : 'total'
			},

			writer: {
				getRecordData: function(record) {
					return { cash_mutation : record.data };
				}
			}
		}
	
  
});

File: app/models/cash_mutation.rb
class CashMutation < ActiveRecord::Base
  belongs_to :cash_bank
  def self.active_objects
    self
  end
  
  def self.create_object(params)
    new_object = self.new 
    new_object.source_class = params[:source_class]
    new_object.source_id = params[:source_id]
    new_object.amount = BigDecimal( params[:amount] || '0')
    new_object.status = params[:status]
    new_object.mutation_date = params[:mutation_date]
    new_object.cash_bank_id = params[:cash_bank_id] 
    new_object.source_code = params[:source_code]
    new_object.save 
    return self 
  end
   
  
  def delete_object
    self.destroy
  end
end
